### PR TITLE
Feature 2 - javadoc, testing, improvements to functionality

### PR DESCRIPTION
**Pull request by yyutiw**
Completed documentation of code throughout and made tests covering all entities, use cases and serialization.

Also made changes to the serialization itself to account for a minor issue where an entity was dependent on another entity, refactored classes for serialization for consistency, and other small edits were made for tests to pass.

Finally integrated all instructor screens, including Task creation, so that should be working now (edit: it is not yet working -- if logged on as an instructor, clicking anything that isn't the New task button and then clicking Cancel/Return will bring you the StudentMain).

(oh and I changed the main screen names back to "StudentMain" and "InstructorMain" so sorry about that mishap)

tldr: my part is basically complete unless I made a dumb mistake; running program as instructor needs work

### CODE REVIEW

**Review by TiareMar — APPROVED** [review]

File: src/main/java/screens/task_management/task_creation_screens/event_creation_screens/EventCreationScreen.java [hunk 1]
```diff
@@ -98,21 +106,26 @@ public void actionPerformed(ActionEvent evt) {
...
+
+                // create the Event
                 eventCreationController.create(title.getText(), valPriority,
                         startDate, endDate, valRecurring, valFrequency);
 
+                // notify user of success and return to main screen
                 showMessageDialog(this, "Event Created Successfully");
-                screenLayout.show(screens, "main");
-            } catch (Exception e) {
+                screenLayout.show(screens, "StudentMain");
```
Comment (TiareMar) [inline src/main/java/screens/task_management/task_creation_screens/event_creation_screens/EventCreationScreen.java:116]: Just one small thing: do task creation screens need to work for both instructors and students?
If so, maybe this screen needs to be initialized with a string for "Student" or "Instructor" from main so that you know whether to go back to the StudentMain or the InstructorMain

Comment (yyutiw) [reply src/main/java/screens/task_management/task_creation_screens/event_creation_screens/EventCreationScreen.java:116]: Hey Tiare, since I made separate Test and Assignment screens for instructors and Events are exclusive to students, I don't think there should be any issues with instructors returning to the student main screen (as long as I connected everything correctly)

**Review by alyson647 — APPROVED** [review]
Looks good! Good documentation throughout your code and there doesn't seem to be any visible violations of clean architecture or solid principles. Really good job with testing your code.

**Review by celinelumqr — APPROVED** [review]
